src/main/java/uk/ac/ebi/ena/webin/cli/upload/FtpService.java

```diff
@@ -40,15 +42,26 @@ public class FtpService implements UploadService {
         try {
             ftpClient.setRemoteVerificationEnabled(false);
             ftpClient.setActivePortRange(40000, 50000);
-            ftpClient.connect(SERVER, FTP_PORT);
-        } catch (IOException e) {
-            throw WebinCliException.systemError(WebinCliMessage.FTP_CONNECT_ERROR.text());
+
+            createRetryTemplate(SocketException.class, IOException.class)
```

Comment:
We could merge createRetryTemplate(...).execute((RetryCallback<Void, Exception>) context -> {

into
```
executeWithRetry((RetryCallback<Void, Exception> ..., Exception ...)
```